### ISSUE TITLE
AR_ColorizeLayersWithGradient has an error, due to the change in the gradient method in C4D 2024.

### ISSUE THREAD

**Issue by Anlv-max** [aturtur/cinema4d-scripts#12]
I have made some adjustments.

```
def ColorizeWithGradient(gradient):
    doc = c4d.documents.GetActiveDocument() # Get the active Cinema 4D document
    doc.StartUndo() # Start recording undos

    irs = render.InitRenderStruct() # Create a rendering structure
    gradientData = gradient.PrepareRenderData(irs) # Initialize gradient data for rendering

    layers = CollectLayers() # Collect layers
    selectedLayers = [l for l in layers if l.GetBit(c4d.BIT_ACTIVE)] # Filter for selected layers

    if len(selectedLayers) == 0: # If no layers are selected
        return

    for i, s in enumerate(selectedLayers):
        doc.AddUndo(c4d.UNDOTYPE_CHANGE, s) # Add undo command
        position = float(i) / (len(selectedLayers) - 1) # Calculate gradient position
        color64 = gradientData.CalcGradientPixel(position) # Calculate gradient color
        colorVector = c4d.Vector(color64.r, color64.g, color64.b) # Convert color
        s[c4d.ID_LAYER_COLOR] = colorVector # Set layer color
        s.Message(c4d.MSG_UPDATE) # Force refresh layer

    doc.EndUndo() # End recording undos
    c4d.EventAdd() # Update Cinema 4D
```

**Comment by aturtur** [comment]
Thanks! Will be fixed in next version

**Comment by aturtur** [comment]
Fixed in v1.77
